projects/ngx-wig/src/lib/config.ts

```diff
@@ -1,14 +1,17 @@
 import { InjectionToken } from '@angular/core';
 import { NgxWigComponent } from './ngx-wig.component';
 
-export type commandFunction = (ctx: NgxWigComponent) => void;
+export type commandFunction = (ctx: NgxWigComponent, valueToSet?: string) => void;
```

Comment:
`valueToSet` parameter - interesting approach. 
could you please elaborate a bit? why ctx is not enough?

Reply:
In the case of the drop-down menu, I did not know how to retrieve the value of the individual button selected and at the same time give the user maximum freedom on which execCommand to launch. It seems to me the simplest way to handle execCommands with two arguments, the second of which is variable. Of course if you know of a more elegant solution I would be grateful for any advice

Reply:
@bampakoa what do you think?

Reply:
Thanks for pinging me @stevermeister 🙏 I think that we can get a handle of the selected toolbar button using the [toolbarButtons](https://github.com/stevermeister/ngx-wig/blob/865c03b84ca528a89a8599f771b439b8325f6973/projects/ngx-wig/src/lib/ngx-wig.component.ts#L62) property. And from there, access the `options` property by filtering the toolbar buttons according to the `command` specified. Something like:

```ts
const selectedButton = ctx.toolbarButtons.find(button => button.command === command);
const value = selectedButton.options;
```

Reply:
Oh, yes. You are right @MPapini91 😃 The word `options` in `execCommand` and in `TButton` confused me 😵

I was wondering if we could format the buttons of the dropdown as `TButton` themselves and have a more abstract structure of:

```
export interface TButton {
  label?: string;
  icon?: string;
  title?: string;
  children?: TButton[];
  command?: string | commandFunction;
  styleClass?: string;
}
```

In that way, we could assume that any button with `children` is a dropdown button and get rid of the hardcoded properties:

```
  isDropdown?: boolean; //isDropdown and visibleDropdown for dropdown menu
  visibleDropdown?: boolean;
```

BTW, I think it is valid to have the same style for the **Font name** and **Headings** buttons as for the **Special characters** button (that is display the select option always and not having the actual dropdown arrow).

Reply:
> I was wondering if we could format the buttons of the dropdown as TButton themselves

I can try this solution (I just need some time), initially I tried to pass each option with the minimum information. But it is interesting and could give more freedom to the user. 

`visibleDropdown`:  I am using it to manage the dropdown display, I don't know if I can find an alternative.

> BTW, I think it is valid to have the same style for the Font name and Headings buttons as for the Special characters button (that is display the select option always and not having the actual dropdown arrow).

Perhaps it is better to eliminate the select version of the dropdown, i.e., this one:

```
<div *ngIf="!button?.isDropdown; else dropdownButton"
  class="select-menu">
  <select (change)="execCommand(button.command, $event.target.value)" [title]="button.title" 
[ngClass]="[button.styleClass]">
              <option value="">{{button.title}}</option>
              <option *ngFor="let option of button.options" [value]="option.value">{{option.label}}</option>
  </select>
</div>
```

It does not work correctly, and seeing a value that is not aligned with the selection point (for example, seeing Times selected where the text is in Arial instead) is likely to confuse users.